### PR TITLE
jwt: possibility to add headers when generating jwt

### PR DESCRIPTION
**Pull request by RodrigoAS28**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [ ] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [x] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
Adding an optional forth parameter to jwt module to add headers.

### CODE REVIEW

**Comment by miconda** [comment]
I see the PR is changing the prototype of ki_jwt_generate() function by adding a new parameter. However that is exported to KEMI and expects the same number of parameters as so far.

You can rename ki_jwt_generate() to ki_jwt_generate_hdrs() and add the new parameter, then re-add ki_jwt_generate() as a wrapper to ki_jwt_generate_hdrs() calling it with the new parameter being NULL.

File: src/modules/jwt/jwt_mod.c [hunk 1]
```diff
@@ -172,18 +175,28 @@ static int jwt_fcache_add(str *key, str *kdata)
  *
  */
 static int ki_jwt_generate(sip_msg_t* msg, str *key, str *alg, str *claims)
+{
+	return ki_jwt_generate_hdrs(msg, key, alg, claims, NULL);
+}
+
```
Comment (miconda) [inline src/modules/jwt/jwt_mod.c:181]: This function has to be moved after `ki_jwt_generate_hdrs()` in order to get it compiled.

**Comment by miconda** [comment]
Looking again over the patch, I didn't notice any free of the parsed params and duplicated string for new headers value. It should be done as it is for claims, or am I missing something?

**Comment by RodrigoAS28** [comment]
> Looking again over the patch, I didn't notice any free of the parsed params and duplicated string for new headers value. It should be done as it is for claims, or am I missing something?

No, you are absolutely right.
Please have another look.

**Pull request merged**